File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/InvitationCodeApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class InvitationCodeApi {
    private final HttpClient client;
    
    public InvitationCodeApi(HttpClient client) {
        this.client = client;
    }

    /** Update an existing invitation code */
    public PlusApiResultInvitationCodeVO update(InvitationCodeForm body) throws Exception {
        return (PlusApiResultInvitationCodeVO) client.put(ApiPaths.backendPath("/invitation/code"), body);
    }

    /** Create a new invitation code */
    public PlusApiResultInvitationCodeVO create(InvitationCodeForm body) throws Exception {
        return (PlusApiResultInvitationCodeVO) client.post(ApiPaths.backendPath("/invitation/code"), body);
    }

    /** Get invitation codes by page */
    public PlusApiResultPageInvitationCodeVO listByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPageInvitationCodeVO) client.post(ApiPaths.backendPath("/invitation/code/list"), body, params);
    }

    /** Get all invitation codes */
    public PlusApiResultListInvitationCodeVO listAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListInvitationCodeVO) client.post(ApiPaths.backendPath("/invitation/code/list/all"), body);
    }

    /** Get an invitation code by ID */
    public PlusApiResultInvitationCodeVO getById(String id) throws Exception {
        return (PlusApiResultInvitationCodeVO) client.get(ApiPaths.backendPath("/invitation/code/" + id + ""));
    }

    /** Delete an invitation code */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/invitation/code/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/InvitationCodeApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class InvitationCodeApi(private val client: HttpClient) {

    /** Update an existing invitation code */
    suspend fun update(body: InvitationCodeForm): PlusApiResultInvitationCodeVO? {
        return client.put(ApiPaths.backendPath("/invitation/code"), body) as? PlusApiResultInvitationCodeVO
    }

    /** Create a new invitation code */
    suspend fun create(body: InvitationCodeForm): PlusApiResultInvitationCodeVO? {
        return client.post(ApiPaths.backendPath("/invitation/code"), body) as? PlusApiResultInvitationCodeVO
    }

    /** Get invitation codes by page */
    suspend fun listByPage(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPageInvitationCodeVO? {
        return client.post(ApiPaths.backendPath("/invitation/code/list"), body, params) as? PlusApiResultPageInvitationCodeVO
    }

    /** Get all invitation codes */
    suspend fun listAllEntities(body: QueryListForm? = null): PlusApiResultListInvitationCodeVO? {
        return client.post(ApiPaths.backendPath("/invitation/code/list/all"), body) as? PlusApiResultListInvitationCodeVO
    }

    /** Get an invitation code by ID */
    suspend fun getById(id: String): PlusApiResultInvitationCodeVO? {
        return client.get(ApiPaths.backendPath("/invitation/code/$id")) as? PlusApiResultInvitationCodeVO
    }

    /** Delete an invitation code */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/invitation/code/$id")) as? PlusApiResultBoolean
    }
}
